### PR TITLE
[bridge] Fix wrong mapping of WorkspaceStatus to prebuild when "snapshot" is missing

### LINKED ISSUE #9676: Finished prebuild does not switch to "available" after execution is finished (stuck in "running")
Note: self-hosted installation

blacked

image

### PR DESCRIPTION
**Pull request by geropl**
## Description
Does what is says in the title, plus adding tests for the mapping. :lotus_position: 

## Related Issue(s)

Fixes #9676

## How to test
 1. `cd components/ws-manager-bridge && yarn test`

## Release Notes

```release-note
fix prebuild state mapping bug in face of missing snapshot (& add tests for it)
```

### CODE REVIEW

**Comment by AlexTugarev** [comment]
@geropl, I've just a single [question](https://github.com/gitpod-io/gitpod/pull/9924#pullrequestreview-969630779) about STOPPED events, other than that, it's a great change!

**Review by AlexTugarev — APPROVED** [review]
/hold

@geropl, I don't want to block this PR just because I don't get the point from the previous comment. Your call, if you think that's OK.

File: components/ws-manager-bridge/src/prebuild-state-mapper.spec.ts
Comment (geropl) [reply components/ws-manager-bridge/src/prebuild-state-mapper.spec.ts:32]: Yes, that mimics the previous behavior, which completely ignored anything except Stopping.

**Pull request merged**
